tests/unit/test_docs.py

```
@@ -0,0 +1,73 @@
+    def test_docs_output_file_created(self, tmp_path):
+        """pdoc produces an HTML file for the neuraldbg module."""
+        subprocess.run(
+            [sys.executable, "-m", "pdoc", "neuraldbg", "--output-dir", str(tmp_path)],
+            capture_output=True,
+        )
+        html_files = list(tmp_path.glob("*.html"))
+        assert len(html_files) >= 1, "No HTML file generated by pdoc"
+
+    def test_all_public_methods_documented(self, tmp_path):
+        """All main public methods of NeuralDbg appear in the generated HTML."""
+        subprocess.run(
```

Ruff (0.15.12)

[error] `subprocess` call: check for execution of untrusted input

(S603)